wagtail/snippets/tests/test_viewset.py

```diff
@@ -769,6 +769,22 @@ def test_boolean_column(self):
         )
 
 
+class TestRelatedFieldColumns(BaseSnippetViewSetTests):
+    model = SnippetChooserModel
+
+    def test_snippetchoosermodel_related_fields(self):
+        advert = Advert(url="https://example.com/free_examples", text="Free Examples")
+        advert.save()
+        ffs = FullFeaturedSnippet(text="test snippet with custom icon")
+        ffs.save()
+        scm = SnippetChooserModel(advert=advert, full_featured=ffs)
+        scm.save()
+
+        response = self.client.get(self.get_url("list"))
+
+        self.assertEqual(response.status_code, 200)
```

Comment:
Could you also assert that the response contains the text of the full featured snippet, as well as the header correctly showing the label (foreign key's verbose_name + the field's verbose_name)

Reply:
Done